### ISSUE TITLE
[M0-Dev] Atributos que va a tener la clase Empleado

### ISSUE THREAD

**Issue by ignaciotitos** [RafaelT00/AutoIV#10]
Que atributos crees que tendrá la clase Empleado?
- Nombre
- Numero ficha de trabajador
- días libres
- proyecto asociado

**Comment by RafaelT00** [comment]
El nombre y la ficha para identificar al trabajador, los demás atributos servirán para poder hacer las asignaciones. Días libres puede ser un vector Vacaciones, ya que un Empleado puede haber tenido varios periodos asignados, además serviría para llevar un fáacil conteo de cuántos días libres ha tenido. Un trabajador puede estar asociado a varios proyectos así que puede ser un vector.
Creo que se pueden añadir también las restricciones como parte del empleado, he pensado también la posibilidad de que los proyectos asociados cuenten como restricciones ya que nos van a restringir en que periodos se le podrá dar vacaciones. 
También creo que hay considerar la antigüedad del empleado en la empresa y su puesto de trabajo.
Si el empleado tiene hijos o no también pienso que es un factor a considerar, al menos para ciertos periodos del año o si son pequeños.
Por último había pensado en una lista de otros factores personales pero creo que sería algo dificil de manejar o representar.

**Comment by ignaciotitos** [comment]
factores personales yo creo que va a ser muy complicado de representar, ya que estos factores pueden indicar más baja que vacaciones en cualquier caso.

**Comment by ignaciotitos** [comment]
Entonces que los proyectos asociados estén en restricciones en vez de empleado?

**Comment by ignaciotitos** [comment]
En cualquier caso puede ser que restricciones se cambie por proyectos asociados y así estos tengan cuanta gente hay en el proyecto, la gente si está de vacaciones o no, etc..

**Comment by ignaciotitos** [comment]
y no entiendo muy bien como vas a llevar el conteo de los días que lleva con un vector de vacaciones

**Comment by RafaelT00** [comment]
> En cualquier caso puede ser que restricciones se cambie por proyectos asociados y así estos tengan cuanta gente hay en el proyecto, la gente si está de vacaciones o no, etc..

Sí, creo que será mejor llevarlo así, ya que la principal restricción son los proyectos en los que está asociado.

> y no entiendo muy bien como vas a llevar el conteo de los días que lleva con un vector de vacaciones

Tenía pensado que cada objeto Vacaciones representara un periodo y recorriendo todo el vector habría contado todos los días que ha estado de vacaciones el empleado.

**Comment by ignaciotitos** [comment]
Mm okey

**Comment by ignaciotitos** [comment]
Lo que todavía no me queda claro es la relación entre proyectos asociados y restricciones. Proyectos asociados va a ser un vector de restricciones o como lo tienes en tu cabeza?

**Comment by RafaelT00** [comment]
Sí, básicamente quiero considerar los proyectos asociados como restricciones.

**Comment by ignaciotitos** [comment]
Vale, aquí queda claro.

**Comment by ignaciotitos** [comment]
Necesito que pongas este issue como inválido
